### PR TITLE
ES6 component conversions

### PR DESCRIPTION
**Pull request by maddiedierker**
Converts the following React components to ES6 syntax:
- EnrollmentUnitAssignment
- ModuleAssignment
- PlcEnrollment

To test that nothing was broken, I ran unit tests locally and did some manual testing, but I'm unfamiliar with this area, so let me know if there's any additional/specific testing I can/should do.

### CODE REVIEW

**Review by clareconstantine — APPROVED** [review]

**Comment by joshlory** [comment]
I vaguely remember case-only renames causing problem in some environments — @davidsbailey does this ring a bell?

**Comment by davidsbailey** [comment]
Nice catch Josh! Yes, this could cause problems. Here's what we tell people to do when it happens during a content scoop. The same rough steps should be sufficient to avoid the problem here too: https://github.com/code-dot-org/code-dot-org/blob/f48e50570dc4604140b8192755efe60e54e5a3ef/bin/content-push#L29-L34

**Comment by maddiedierker** [comment]
thank you, @joshlory and @davidsbailey! should i perform these steps on my branch before merging, or at another point in the deploy process?

**Comment by davidsbailey** [comment]
You should perform these steps on your branch before merging. The rule to follow is, no individual commit in the staging branch can contain a case-only rename.

File: apps/src/code-studio/pd/professional_learning_landing/plcElements/EnrollmentUnitAssignment.jsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, {PropTypes} from 'react';
 import color from '../../../../util/color';
-import ModuleAssignment from './moduleAssignment';
+import ModuleAssignment from './ModuleAssignment';
```
Comment (Erin007) [inline apps/src/code-studio/pd/professional_learning_landing/plcElements/EnrollmentUnitAssignment.jsx:3]: yay! The lowercase and snake_case components and files in PLC land bug me more than they should 🙂 - this is a great start to making them more consistent with the rest of our component names.

**Review by Erin007 — APPROVED** [review]
Thanks, Maddie!

**Comment by Erin007** [comment]
FWIW I've made casing-only commits directly through the Github UI, and they've worked. I've only had problems when I make the change in my editor then push 🤷🏼‍♀️ [example](https://github.com/code-dot-org/code-dot-org/pull/26326/commits/c076e20f2b1766386153c59f8cb736148891ea20)

**Comment by davidsbailey** [comment]
IIRC the extra steps avoid problems when OS X developers who do not have `git config core.ignorecase` later pull down your changes. I could be mis-remembering, but for now I'd rather follow the steps baked into our process in hopes of not tripping up any developer in that situation.

**Pull request merged**